libticonv/trunk/src/iconv.c

```diff
@@ -61,10 +61,10 @@ TIEXPORT4 ticonv_iconv_t TICALL ticonv_iconv_open (const char *tocode, const cha
 /* Convert at most *INBYTESLEFT bytes from *INBUF according to the
    code conversion algorithm specified by CD and place up to
    *OUTBYTESLEFT bytes in buffer at *OUTBUF.  */
-TIEXPORT4 size_t TICALL ticonv_iconv (ticonv_iconv_t cd, char ** restrict inbuf,
```

Comment:
Meh. This unconditionally reintroduces a previously long-standing known portability issue I fixed nearly 7 years ago in 933b4c66a0245ad5c1168e21b4cdcc581973fe86 .
Whatever the autotools build system does when configure.ac contains `AC_C_RESTRICT` - IIRC, something along the lines of replacing `restrict` by `__restrict` on a number of platforms - made libticonv cross-buildable for Win32 using the MinGW toolchain. I guess that the headers used by the MinGW toolchain - which must be drawing partially from MS headers - do not have the internal conflict with the Windows SDK you mention in the commit message ?
At least, this should be conditional on e.g. MSC_VER or the availability of the offending Windows SDK, right ?

Reply:
It looks like the following CMake test allows to conditionalize it:

```cmake
if(NOT c_restrict IN_LIST CMAKE_C_COMPILE_FEATURES)
    add_compile_definitions(restrict=__restrict)
endif()
```

Reply:
Actually nevermind, it's a bit more involved, but it should be doable.

Reply:
Nevermind, I see no way of making it work. `__restrict` seems to be the way to use on clang, gcc, and MSVC.
I suggest that if anything else is required somewhere, a #define be made.